### PR TITLE
Add Payment Request IDL tests

### PR DESCRIPTION
**Pull request by sideshowbarker**
Intends to fully test IDL conformance of implementations of all interfaces in [Payment Request spec](https://w3c.github.io/browser-payment-api/):
- [PaymentRequest interface](https://w3c.github.io/browser-payment-api/#paymentrequest-interface)
- [PaymentResponse interface](https://w3c.github.io/browser-payment-api/#paymentresponse-interface)
- [PaymentAddress interface](https://w3c.github.io/browser-payment-api/#paymentaddress-interface)

### CODE REVIEW

File: payment-request/interfaces.html [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+<script src=/resources/testharnessreport.js></script>
+<script src=/resources/WebIDLParser.js></script>
+<script src=/resources/idlharness.js></script>
+<script type=text/plain class=untested>
+interface EventHandler {};
+interface Event {};
+interface EventInit {};
+interface EventTarget {};
+</script>
+<script type=text/plain>
```
Comment (halindrome) [inline payment-request/interfaces.html]: I know this is a little pedantic, but put text/plain in quotes?

**Comment by halindrome** [comment]
If you can make the lint recommended change from the build failure (trailing whitespace) and resubmit, I will check it again.

**Comment by sideshowbarker** [comment]
> If you can make the lint recommended change from the build failure (trailing whitespace) and resubmit, I will check it again.

OK, see the latest commit. Note that it also renames the test file to `interfaces.https.html`, per the filename convention we use for test files that must be run under TLS rather than insecure HTTP.

**Comment by sideshowbarker** [comment]
FYI the test failures for this in Chrome are known issues with Blink’s WebIDL conformance that have already been reported for `idlharness.js` tests for other APIs implemented in Chrome:

* [Bug: IDL bindings: Calling a promise-returning method with this=null should reject, not throw](https://bugs.chromium.org/p/chromium/issues/detail?id=627309&desc=2)
* [Bug: class string of Type.prototype expected "[object TypePrototype]" but got "[object Type]"](https://bugs.chromium.org/p/chromium/issues/detail?id=635694)

**Comment by halindrome** [comment]
LGTM.   Unfortunate about the bug in Blink...

**Pull request merged**